add docs for making context type-safe (#2418)
graphile/crystal: grafast/website/grafast/step-library/standard-steps/context.md

  was an object
- `.at(index)` - gets the value at index `index` assuming the parsed JSON value
  was an array

> ## Typescript

Reviewer: ```suggestion
## TypeScript
```

> Typescript declaration merging should be used to extend the default grafast context and make the usage of this step type safe in the plan resolvers. Here's an example:

Reviewer: ```suggestion
TypeScript declaration merging should be used to detail the properties you are
making available on GraphQL context such that usage of this step is type safe in
plan resolvers. For example:
```

```ts
declare global {
  namespace Grafast {
    interface Context {

>       postgresClient: PostgresClient;

Reviewer: ```suggestion
      currentUserId?: number;
```

    };
  }
}
```

> The code above would make accessing `context().get("postgresClient")` and its return value type-safe in the plan resolvers.

Reviewer: ```suggestion
The code above would mean that `context().get("currentUserId")` returns
`Step<number | undefined>`, thereby making its usage in plan resolvers type
safe.
```

  was an object
- `.at(index)` - gets the value at index `index` assuming the parsed JSON value
  was an array

## TypeScript

TypeScript declaration merging should be used to detail the properties you are
making available on GraphQL context such that usage of this step is type safe in
plan resolvers. For example:

```ts
declare global {
  namespace Grafast {
    interface Context {
      currentUserId?: number;

>     };

Reviewer: ```suggestion
    }
```
